ansible/rancher/rancher-playbook.yml

```diff
@@ -0,0 +1,101 @@
+    terraform_workspace: "{{ lookup('env', 'TF_WORKSPACE') }}"
+    # Construct the path to the workspace's state file
+    terraform_state_file: "../../terraform/aws/cluster_nodes/terraform.tfstate.d/{{ terraform_workspace }}/terraform.tfstate"
+    fqdn: "{{ lookup('cloud.terraform.tf_output', 'fqdn', state_file=terraform_state_file) }}"
+  tasks:
+    - name: Check if kubeconfig file exists
+      stat:
+        path: "{{ kubeconfig_file }}"
+      register: kubeconfig_file_stat
+
+    - name: Fail if kubeconfig file does not exist
+      fail:
+        msg: "kubeconfig file '{{ kubeconfig_file }}' does not exist."
+      when: not kubeconfig_file_stat.stat.exists
+
+    - name: Install cert-manager CRDs
+      kubernetes.core.k8s:
+        kubeconfig: "{{ kubeconfig_file }}"
+        src: https://github.com/cert-manager/cert-manager/releases/download/v{{ cert_manager_version }}/cert-manager.crds.yaml
+        state: present
+      when: cert_manager_version != ""
+
+    - name: Install cert-manager
+      kubernetes.core.k8s:
+        kubeconfig: "{{ kubeconfig_file }}"
+        src: https://github.com/cert-manager/cert-manager/releases/download/v{{ cert_manager_version }}/cert-manager.yaml
+        state: present
+        wait: true
+        wait_timeout: 300
+      when: cert_manager_version != ""
+
+    - name: Wait for cert-manager to be ready
+      kubernetes.core.k8s_info:
+        kubeconfig: "{{ kubeconfig_file }}"
+        api_version: apps/v1
+        kind: Deployment
+        namespace: cert-manager
+        name: cert-manager
+      register: cert_manager_deployment
+      until: cert_manager_deployment.resources[0].status.readyReplicas == cert_manager_deployment.resources[0].status.replicas
+      retries: 10
+      delay: 30
+      when: cert_manager_version != ""
+
+    - name: Add Rancher Helm repository
+      kubernetes.core.helm_repository:
+        kubeconfig: "{{ kubeconfig_file }}"
+        name: "{{ rancher_chart_repo }}"
+        repo_url: "{{ rancher_chart_repo_url }}"
+      when: rancher_version != ""
+
+    - name: Install Rancher
+      kubernetes.core.helm:
+        kubeconfig: "{{ kubeconfig_file }}"
+        name: rancher
+        chart_ref: "{{ rancher_chart_repo }}/rancher"
+        release_namespace: cattle-system
+        create_namespace: true
+        chart_version: "{{ rancher_version }}"
+        wait: true
+        wait_timeout: "600s"
+        values:
+          hostname: "{{ fqdn }}"
+          bootstrapPassword: "{{ bootstrap_password }}"
+      when: rancher_version != ""
+
+    - name: Wait for Rancher to be ready
+      kubernetes.core.k8s_info:
+        kubeconfig: "{{ kubeconfig_file }}"
+        api_version: apps/v1
+        kind: Deployment
+        namespace: cattle-system
+        name: rancher
+      register: rancher_deployment
+      until: rancher_deployment.resources[0].status.readyReplicas == rancher_deployment.resources[0].status.replicas
```

Comment:
Should we add a check to see if the healthcheck endpoints of the ingress is in a good state? https://ranchermanager.docs.rancher.com/how-to-guides/new-user-guides/infrastructure-setup/amazon-elb-load-balancer#health-check-paths-for-nginx-ingress-and-traefik-ingresses

Reply:
Should be as simple as `curl {{fqdn}/ping` or `curl {{fqdn}}/healthz` and making sure the response status is 200 I guess. Not sure if that will work that easily but if its handled elsewhere then its probably not needed

Reply:
Actually I guess it would be after the "Display Rancher URL" step with `curl https://{{ rancher_ingress.resources[0].spec.rules[0].host }}/[ping || healthz]`

Reply:
aaah yes I can do that. Let me try, before I merge the PR.

Reply:
From the upstream documentation:

> Note:
> Ingress is frozen. New features are being added to the [Gateway API](https://kubernetes.io/docs/concepts/services-networking/gateway/).

https://kubernetes.io/docs/concepts/services-networking/ingress 

So, even if we added this condition, we may have to update it soon. In my opinion it's not really worth to add since it's lifecycle gonna be relatively short